Create and destroy model per test

johnsca opened the pull request:
This also includes @bcsaller's TUIView improvements.

petevg commented on matrix/tasks/glitch/main.py:
```diff
@@ -104,9 +104,12 @@ def default_resolver(model, kind, name):
 
-        await actionf(rule, model, objects, **action)
+        try:
+            await asyncio.wait_for(actionf(rule, model, objects, **action), 30)
```
After we've defined success and failure better, as per https://github.com/juju-solutions/matrix/issues/41, we should make this fail, I think, as timeouts are generally indicative of bugs.

... or maybe we have a "test core" config param, and if that is true, then timeouts lead to failure.

In any case, I'm okay with this making it into the codebase as is for now.

petevg commented:
This looks good, and runs well against a local controller on my laptop.

One minor nitpick: it would be nice to have a "model destroyed" message at the very end. As it was, the last message in my log was "destroying model ...", and I thought that it had hung.

bcsaller commented on matrix/rules.py:
```diff
@@ -340,7 +355,12 @@ def allow_event(e):
+                await self.add_model(context)
+                await self.run_once(context, test)
+            finally:
+                await self.destroy_model(context)
```
We may want to honor a cmdline flag to bail here before cleaning the model so the developer can check the remaining units in more detail

johnsca commented:
@petevg Technically, the model is not destroyed when that API call returns, it has just sent the request off to the controller.  I added a log message to indicate that it returned, at least, but do you think we should actually block until the model is gone to verify that it was successfully removed?

@bcsaller I added a `--keep-models` flag, and commented out the timeline action for now.

petevg commented:
@johnsca Works great. Looks good. +1 (I don't think that we need to block to make sure the model is destroyed for now -- the user can clean up old models if matrix leaves a mess every once in a while.)
